### ISSUE TITLE
Archive Failed with Exception

### ISSUE THREAD

**Issue by penguin02007** [github/gh-gei#838]
## Description

When the archive step failed, an exception is raise instead of continue to the next migration. While it is helpful to see the error, I feel it is more important for the migration to continue.

```
[14:54] [INFO] Waiting for archive with id 2096 generation to finish. Current status: failed
[14:54] [ERROR] OctoshiftCLI.OctoshiftCliException: Archive generation failed for id: 2096
   at OctoshiftCLI.GithubEnterpriseImporter.Handlers.MigrateRepoCommandHandler.WaitForArchiveGeneration(GithubApi githubApi, String githubSourceOrg, Int32 archiveId)
   at OctoshiftCLI.GithubEnterpriseImporter.Handlers.MigrateRepoCommandHandler.GenerateAndUploadArchive(String githubSourceOrg, String sourceRepo, String awsBucketName, Boolean skipReleases, Boolean lockSourceRepo, Boolean blobCredentialsRequired)
   at OctoshiftCLI.GithubEnterpriseImporter.Handlers.MigrateRepoCommandHandler.Handle(MigrateRepoCommandArgs args)
   at OctoshiftCLI.Extensions.CommandExtensions.<>c__DisplayClass1_0`3.<<ConfigureCommand>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext )
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()
   ```

https://github.com/github/gh-gei/blob/main/src/gei/Handlers/MigrateRepoCommandHandler.cs#L139

## Reproduction Steps
1. `gh gei generate-script ... --output migrate.ps1`
2.  `./migrate.ps1`

**Comment by timrogers** [comment]
@penguin02007 Thanks for your contribution! I'd like to understand the problem here a bit better - are you running multiple migrations using a script from `gh gei generate-script`, and then finding that the script exits if archive generation fails for a single repo?

**Comment by penguin02007** [comment]
Yes exactly.

**Comment by timrogers** [comment]
@penguin02007 Thanks!

**Comment by dylan-smith** [comment]
We've talked about this problem a bit in the past (here's a related issue: https://github.com/github/gh-gei/issues/781).

We'd like to solve it by making the script more robust.  The PR you submitted is problematic because if you don't raise an exception the CLI will exit with exit code 0 (aka success), and any script calling the CLI will not be able to easily detect whether the command failed or succeeded.

We'd rather make the generated script more robust where a failed command doesn't cause the entire script to halt. We have tried to do that already, but obviously there's some flaws in the script logic.

**Comment by ArinGhazarian** [comment]
Closing as this was completed in #842

**Comment by spideyz0r** [comment]
It looks like this wasn't fixed, I'm using v1.8.0 and the migration halts:
```
[2024-06-14 15:45:58] [INFO] Waiting for archive with id 35 generation to finish. Current status: exporting
[2024-06-14 15:46:08] [INFO] Waiting for archive with id 35 generation to finish. Current status: exporting
[2024-06-14 15:46:18] [INFO] Waiting for archive with id 35 generation to finish. Current status: failed
[2024-06-14 15:46:18] [ERROR] Archive generation failed for id: 35 
```

```
 $ gh gei --version
[2024-06-14 15:48:07] [INFO] You are running an up-to-date version of the gei CLI [v1.8.0]
1.8.0
```

Is this the same problem?

**Comment by ArinGhazarian** [comment]
👋 @spideyz0r,

Are you running multiple migration or just one? If you're running a single migration then without an archive the migration cannot continue and it's the normal behavior. The problem that we solved in #842 was for migrating multiple repos so if we fail to queue a single migration we don't halt the entire migration script and continue queuing up the rest.
